tools/autograd/derivatives.yaml

```diff
@@ -1503,6 +1503,11 @@
   grad_output: grad.slice_symint(dim, start, end, step)
   result: auto_linear
 
+- name: slice_inverse(Tensor(a) self, Tensor src, int dim=0, SymInt? start=None, SymInt? end=None, SymInt step=1) -> Tensor(a)
+  self: grad.slice_symint(dim, start, end, step)
+  src: slice_scatter_symint(grad, zeros_like(self), dim, start, end, step)
+  result: auto_linear
```

Comment:
Maybe add a small comment about why we need this formula manually

Reply:
it's because it has to be CompositeExplicit so we can override it for a subclass